### PR TITLE
New user

### PR DESCRIPTION
**Pull request by NSMcEnroe**
## Description
This PR includes everything we worked on:

- /register
-New actions in user controller
-New views for show and new
- New tests for registering a user

This does not include sad paths and edge cases.  I'll work on those today.

### Type of Change

- [x] **fix**
- [x] **feat**
- [x] **test**
- [ ] **refactor**
- [ ] **docs**

### Checklist

- [x] **Code has been self reviewed**
- [ ] **Code runs without any errors**
- [ ] **Thorough testing has been implemented if adding feature**
- [ ] **All tests pass**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -6,4 +8,29 @@ def index
...
+    @user = User.new
+  end
+
+  def create
+    user = User.new(user_params)
+
+    if !user.save
+      # Email is not unique, show flash alert
+      flash[:alert] = "Please try again.  All fields must be filled in and the emails adress must be unique."
+      render 'new'
```
Comment (ttakahashi1591) [inline app/controllers/users_controller.rb:22]: This looks good adding the flash message for `Emails` that are not unique and ensuring that all fields are filled out prior to hitting the `Register` button.

File: app/models/user.rb [hunk 2]
```diff
@@ -7,5 +7,5 @@ class User < ApplicationRecord
   has_many :viewing_parties, through: :user_viewing_parties
 
   validates :name, presence: true
-  validates :email, presence: true
+  validates :email, presence: true, uniqueness: true
```
Comment (ttakahashi1591) [inline app/models/user.rb:10]: Goos idea adding the uniqueness check in our `User` model validations.

**Review by ttakahashi1591 — COMMENTED** [review]
This all looks good and thank you for your thoughts around refactoring to add sad paths and edgecases!
